src/sentry/api/endpoints/assistant.py

```diff
@@ -0,0 +1,89 @@
+from __future__ import absolute_import
+
+from datetime import timedelta
+from django.http import HttpResponse
+from django.utils import timezone
+from rest_framework import serializers
+from rest_framework.permissions import IsAuthenticated
+from rest_framework.response import Response
+
+from sentry.api.base import Endpoint
+from sentry.assistant.guides import GUIDES
+from sentry.models import AssistantActivity
+
+
+VALID_GUIDE_IDS = frozenset(v['id'] for v in GUIDES.values())
+VALID_STATUSES = frozenset(('viewed', 'dismissed', 'snoozed'))
+VALID_SNOOZE_DURATION_HOURS = frozenset((1, 24, 24 * 7))
+
+
+class AssistantSerializer(serializers.Serializer):
+    guide_id = serializers.ChoiceField(
+        choices=zip(VALID_GUIDE_IDS, VALID_GUIDE_IDS),
+        required=True,
+    )
+    status = serializers.ChoiceField(
+        choices=zip(VALID_STATUSES, VALID_STATUSES),
+        required=True,
+    )
+    duration_hours = serializers.ChoiceField(
+        choices=zip(VALID_SNOOZE_DURATION_HOURS, VALID_SNOOZE_DURATION_HOURS),
+    )
+    useful = serializers.BooleanField()
+
+    def validate(self, data):
+        if (data['status'] == 'snoozed' and
+                data.get('duration_hours') not in VALID_SNOOZE_DURATION_HOURS):
+            raise serializers.ValidationError("must specify a valid snooze duration")
+        return data
+
+
+class AssistantEndpoint(Endpoint):
+    permission_classes = (IsAuthenticated, )
+
+    def get(self, request):
+        """Return all the guides the user has not seen, dismissed, or snoozed."""
+        exclude = AssistantActivity.objects.filter(
+            user=request.user,
+        ).exclude(
+            snoozed_until_ts__lt=timezone.now(),
+        )
+        exclude_ids = set(e.id for e in exclude)
```

Comment:
This can be written as:

```python
exclude_ids = set(AssistantActivity.objects.filter(
    user=request.user,
).exclude(
    snoozed_until_ts__lt=timezone.now(),
).values_list('id', flat=True))
```

The key thing to point out is the `values_list` change which tells the ORM to only return that one column, and not full `AssistantActivity` objects, which you then only use the `id` from.